src/plugins/intel_cpu/src/transformations/snippets/aarch64/op/gemm_cpu.cpp

```diff
@@ -61,10 +61,30 @@ void GemmCPU::validate_and_infer_types() {
 
 void GemmCPU::validate_element_type(const ov::element::Type& type_0, const ov::element::Type& type_1) {
     OPENVINO_ASSERT(
-        all_of(type_0, type_1, element::f32),
+        any_of(type_0, element::f32, element::f16) && any_of(type_1, element::f32, element::f16) && type_0 == type_1,
         "GemmCPU doesn't support element type in0:" + type_0.get_type_name() + " in1:" + type_1.get_type_name());
 }
 
+ov::element::Type GemmCPU::get_output_type() const {
+    const auto in_type0 = get_input_element_type(0);
+    const auto in_type1 = get_input_element_type(1);
+
+    OPENVINO_ASSERT(in_type0 == in_type1,
+                    "GemmCPU expects matching input types, got: ",
+                    in_type0.get_type_name(),
+                    " and ",
+                    in_type1.get_type_name());
+
+    if (in_type0 == element::f16) {
+        return element::f16;
+    }
+    if (in_type0 == element::f32) {
+        return element::f32;
+    }
```

Comment:
Can we just return `get_input_element_type(0);`? If you want to additionally validate input element types, you can just call `validate_element_type`